src/viam/app/app_client.py

```diff
@@ -1200,6 +1394,50 @@ async def list_modules(self) -> List[Module]:
         response: ListModulesResponse = await self._app_client.ListModules(request, metadata=self._metadata)
         return list(response.modules)
 
-    # TODO: implement
-    async def create_key(self, authorizations: List[Authorization], name: str) -> Tuple[str, str]:
-        raise NotImplementedError()
+    # TODO(RSDK-5569): when user-based auth exists, make `name` default to `None` and let
+    # app deal with setting a default.
+    async def create_key(self, authorizations: List[APIKeyAuthorization], name: Optional[str] = None) -> Tuple[str, str]:
+        """Creates a new API key.
+
+        Args:
+            authorizations (List[viam.proto.app.Authorization]): A list of authorizations to associate
+            with the key.
+            name (Optional[str]): A name for the key. If None, defaults to the current timestamp.
+
+        Raises:
+            GRPCError: If the authorizations list is empty.
+
+        Returns:
+            Tuple[str, str]: The api key and api key ID.
+        """
+        name = name if name else str(datetime.now())
```

Comment:
same-ish Q as noted above

```suggestion
        name = name if name is not None else str(datetime.now())
```